libraries/Microsoft.Bot.Builder.Azure/CosmosDbStorage.cs

```diff
@@ -39,13 +39,27 @@ public class CosmosDbStorage : IStorage
         /// using the provided CosmosDB credentials, database ID, and collection ID.
         /// </summary>
         /// <param name="cosmosDbStorageOptions">Cosmos DB storage configuration options.</param>
-        public CosmosDbStorage(CosmosDbStorageOptions cosmosDbStorageOptions)
+        /// <param name="jsonSerializer">Optional JsonSerializer. Mind that the TypeNameHandling, NullvalueHandling and ContractResolver will be overridden.</param>
+        public CosmosDbStorage(CosmosDbStorageOptions cosmosDbStorageOptions, JsonSerializer jsonSerializer = null)
```

Comment:
We can't take this, as is, as it breaking binary compat. For this approach, you need add an additional ctor that takes the JsonSerializer.